src/anndata/compat/__init__.py

```diff
@@ -15,18 +15,35 @@
 import h5py
 import numpy as np
 import pandas as pd
+import scipy
+import scipy.sparse
 from packaging.version import Version
-from scipy.sparse import issparse, spmatrix
 
 from .exceptiongroups import add_note  # noqa: F401
 
+#############################
+# scipy sparse array comapt #
+#############################
+
+
+CAN_USE_SPARSE_ARRAY = Version(scipy.__version__) >= Version("1.11")
+
+if not CAN_USE_SPARSE_ARRAY:
+
+    class SpArray:
+        @staticmethod
+        def __repr__():
+            return "mock scipy.sparse.sparray"
+else:
+    SpArray = scipy.sparse.sparray
+
```

Comment:
Well, the lower-scipy-version miss makes sense if the repr is not called, but not sure what's up with the second half

Reply:
In reference to codecov